### ISSUE TITLE
Proposal: I/O for simple access values

### ISSUE THREAD

**Issue by okellogg** [persan/auto-io-gen#2]
Years ago I had a brief discussion with [Stephen Leake](https://github.com/stephe-ada-guru) about supporting access values in Auto_Text_IO.
Finally I found a little time to get back to the idea.
I am attaching a small demo of what it could look like.

The textual notation for access values breaks compatibility with the Ada syntax, i.e. you cannot copy/paste the textual output verbatim into an Ada program and compile it (as is the case for Auto_Text_IO without access values).
This is due to hexadecimal numbers printed as identifiers for the access values. The identifiers are required for reconstructing arbitrary graphs of linked structures (such as circular buffers) where it must be avoided to print the content of a single access value multiple times.

By "simple" access values I mean those values for which one can write "new T" without providing parameters. I.e. accesses to unconstrained/indefinite types are not supported.

Would this be of interest?

[access_io_demo.tar.gz](https://github.com/persan/auto-io-gen/files/4398019/access_io_demo.tar.gz)

**Comment by okellogg** [comment]
I am working on this in the branch [access_type_support](https://github.com/okellogg/auto-io-gen/tree/access_type_support) of my fork.
Note however that it's very early in the development, the generated code does not compile yet.

**Comment by okellogg** [comment]
In the description I wrote:

> This is due to hexadecimal numbers printed as identifiers for the access values.

TLDR: The identifiers are now printed in decimal.

Long explanation:

It turns out that using the access value as the ID (via unchecked conversion to System.Storage_Elements.Integer_Address) is not good for getting repeatable results.
The pointer values (addresses) will usually be different from run to run when `Put`ting them.
This is because the addresses provided by the heap allocator are not always the same for the same `new` statements.
To get repeatable IDs for the access values, a separate global ID counter is now used.
The counter will be incremented for each unique access value written.

Here is an example on basis of a linked list structure.

Previous format - notice that the hex access identifiers change from run to run:
```
  #0000000000EE32A0'
   (Data => ...., Prev => null,
    Next => #0000000000EE32C0'
             (Data => ...., Prev => ^0000000000EE32A0, Next => null))
```

New format - here the access identifiers are always 1, 2, 3, ...
```
  #1'
   (Data => ...., Prev => null,
    Next => #2'
             (Data => ...., Prev => ^1, Next => null))
```
